include/lsst/afw/image/PhotoCalib.h

```diff
@@ -0,0 +1,426 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_IMAGE_PHOTOCALIB_H
+#define LSST_AFW_IMAGE_PHOTOCALIB_H
+
+/**
+ * The photometric calibration of an exposure.
+ *
+ * A PhotoCalib is a BoundedField (a function with a specified domain) that converts between calibrated
+ * counts-on-chip (ADU) to flux and magnitude. It is defined in terms of "maggies", which are a linear
+ * unit defined in SDSS: http://www.sdss.org/dr12/algorithms/magnitudes/#nmgy
+ *
+ * PhotoCalib is immutable.
+ *
+ * The spatially varying flux/magnitude zero point is defined such that,
+ * at a position (x,y) in the domain of the boundedField zeroPoint
+ * and for a given measured source counts:
+ * @f[
+ *     counts / zeroPoint(x,y) = flux (in maggies)
+ * @f]
+ * while the errors (constant on the domain) are defined as:
+ * @f[
+ *     sqrt((countsSigma/counts)^2 + (zeroPointSigma/zeroPoint)^2)*flux = fluxSigma (in maggies)
+ * @f]
+ * This implies that the conversions from counts and counts error to magnitude and magnitude error
+ * are as follows:
+ * @f[
+ *     -2.5 * log_10(counts / zeroPoint(x,y)) = magnitude
+ * @f]
+ * and
+ * @f[
+ *     2.5*log(10) * sqrt((countsSigma/counts)^2 + (zeroPointSigma/zeroPoint)^2) = magSigma
```

Comment:
Thanks for checking this! I did indeed mis-translate that from my mathematica notebook. Good catch! Fixed in 3 places (test, docstring, code).